src/test/regress/spec/isolation_metadata_sync_vs_all.spec

```diff
@@ -0,0 +1,197 @@
+
+	CREATE TABLE dist_table (x int, y int);
+	SELECT create_distributed_table('dist_table', 'x');
+
+	CREATE TABLE dist_partitioned_table (x int, y int) PARTITION BY RANGE(y);
+	SELECT create_distributed_table('dist_partitioned_table', 'x');
+
+	CREATE TABLE dist_partitioned_table_p1(x int, y int);
+
+	-- Create the necessary test utility function
+	CREATE OR REPLACE FUNCTION trigger_metadata_sync()
+		RETURNS void
+		LANGUAGE C STRICT
+		AS 'citus';
+
+	CREATE OR REPLACE FUNCTION activate_node_snapshot()
+		RETURNS text[]
+		LANGUAGE C STRICT
+		AS 'citus';
+	SELECT create_distributed_function('activate_node_snapshot()');
+}
+
+// ensure neither node's added for the remaining of the isolation tests
+teardown
+{
+	DROP TABLE IF EXISTS ref_table,
+						 dist_table,
+						 dist_partitioned_table,
+						 dist_partitioned_table_p1,
+						 dist_partitioned_table_p2,
+						 new_dist_table,
+						 new_ref_table;
+
+	DROP FUNCTION activate_node_snapshot();
+
+	SELECT master_remove_node(nodename, nodeport) FROM pg_dist_node;
+}
+
+session "s1"
+
+step "s1-begin"
+{
+    BEGIN;
+}
+
+step "s1-commit"
+{
+    COMMIT;
+}
+
+step "s1-rollback"
+{
+    ROLLBACK;
+}
+
+step "s1-start-metadata-sync"
+{
+	SELECT start_metadata_sync_to_node('localhost', 57638);
+}
+
+session "s2"
+
+step "s2-begin"
+{
+	BEGIN;
+}
+
+step "s2-commit"
+{
+	COMMIT;
+}
+
+step "s2-rollback"
+{
+	ROLLBACK;
+}
+
+step "s2-alter-table"
+{
+	ALTER TABLE dist_table ADD COLUMN z int;
+}
+
+step "s2-add-fk"
+{
+	ALTER TABLE dist_table ADD CONSTRAINT y_fk FOREIGN KEY (y) REFERENCES ref_table(y);
+}
+
+step "s2-drop-fk"
+{
+	ALTER TABLE dist_table DROP CONSTRAINT y_fk;
+}
+
+step "s2-drop-table"
+{
+	DROP TABLE dist_table;
+}
+
+step "s2-create-dist-table"
+{
+	CREATE TABLE new_dist_table(id int, data int);
+	SELECT create_distributed_table('new_dist_table', 'id');
+}
+
+step "s2-create-ref-table"
+{
+	CREATE TABLE new_ref_table(id int, data int);
+	SELECT create_reference_table('new_ref_table');
+}
+
+step "s2-attach-partition"
+{
+	ALTER TABLE dist_partitioned_table ATTACH PARTITION dist_partitioned_table_p1 FOR VALUES FROM (1) TO (9);
+}
+
+step "s2-detach-partition"
+{
+	ALTER TABLE dist_partitioned_table DETACH PARTITION dist_partitioned_table_p1;
+}
+
+step "s2-create-partition-of"
+{
+	CREATE TABLE dist_partitioned_table_p2 PARTITION OF dist_partitioned_table FOR VALUES FROM (10) TO (20);
+}
+
+step "s2-create-type"
+{
+	CREATE TYPE my_type AS (a int, b int);
+}
+
+step "s2-drop-type"
+{
+	DROP TYPE my_type;
+}
+
+step "s2-create-dist-func"
+{
+	CREATE FUNCTION squares(int) RETURNS SETOF RECORD
+    AS $$ SELECT i, i * i FROM generate_series(1, $1) i $$
+    LANGUAGE SQL;
+
+	SELECT create_distributed_function('squares(int)');
+}
+
+step "s2-drop-dist-func"
+{
+	DROP FUNCTION squares(int);
+}
+
+session "s3"
+
+step "s3-compare-snapshot"
+{
+	SELECT bool_and(result::text[] @> activate_node_snapshot() AND result::text[] <@ activate_node_snapshot()) AS same_snapshot_across_cluster
```

Comment:
Not sure if my below suggestion is simpler than this, but at least I find it easier to read and seems does a full line by line comparison across all nodes:
```SQL

SELECT count(*) = 0
FROM 
(
    (
    	SELECT unnest(activate_node_snapshot())
     		EXCEPT 
     	SELECT unnest(string_to_array(RESULT, 'CANNOTBEUSEDSTRING')) AS unnested_result
     	FROM run_command_on_workers($$SELECT array_to_string(activate_node_snapshot(), 'CANNOTBEUSEDSTRING')$$)
     	GROUP BY unnested_result
     ) 
  UNION
    (
    	SELECT unnest(string_to_array(RESULT, 'CANNOTBEUSEDSTRING')) AS unnested_result
     	FROM run_command_on_workers($$SELECT array_to_string(activate_node_snapshot(), 'CANNOTBEUSEDSTRING')$$)
     	GROUP BY unnested_result
  			EXCEPT
    	SELECT unnest(activate_node_snapshot())
    )
) AS foo;
```

One difference I found is that the coordinator generates more lines for things like:
```SQL
 SET citus.enable_ddl_propagation TO 'on'
 SET citus.enable_ddl_propagation TO 'off'
 SET ROLE onderkalaci
```

I think that is a separate discussion, the above query still does a full control for

Reply:
I simplified your suggested queries. Let me know if you think they are ok.

Basically, I removed conversions from text[] array into string, and vice-versa. We can already use text[] values returned by the `run_command_on_workers()` UDF.